File: 1512-number-of-good-pairs/1512-number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        n=len(nums)
        count={}
        ans=0
        for i in range(n):
            g=count.get(nums[i],[])
            for m in g:
                ans+=1
            count[nums[i]]=g+[i]
        return ans

File: 1512-number-of-good-pairs/1512-number-of-good-pairs.cpp
class Solution {
public:
    int numIdenticalPairs(vector<int>& nums) {
        int goodPairs=0;
        for(int i: unordered_set(nums.begin(), nums.end())){
            int n = count(nums.begin(), nums.end(), i);
            goodPairs+=n*(n-1)/2;
        }
        return goodPairs;
    }
};